src/Data/Pool/Internal.hs

```diff
@@ -87,20 +96,32 @@ newPool pc = do
     error "poolCacheTTL must be at least 0.5"
   when (poolMaxResources pc < 1) $ do
     error "poolMaxResources must be at least 1"
-  numStripes <- getNumCapabilities
-  when (numStripes < 1) $ do
+  numStripesRequested <- maybe getNumCapabilities pure (poolNumStripes pc)
+  when (numStripesRequested < 1) $ do
     error "numStripes must be at least 1"
-  pools <- fmap (smallArrayFromListN numStripes) . forM [1..numStripes] $ \n -> do
+  let stripeResourceAllocation =
+        howManyStripes Input
+          { inputMaxResources = poolMaxResources pc
+          , inputStripes = numStripesRequested
+          }
+      stripeAllocations =
+        robin stripeResourceAllocation
+      allocationWithIndex =
+        zip [1..] stripeAllocations
+      numStripes =
+        allowedStripes stripeResourceAllocation
+
+  pools <- fmap (smallArrayFromListN numStripes) . forM allocationWithIndex $ \(n, i) -> do
```

Comment:
It's swapped?

```suggestion
  pools <- fmap (smallArrayFromListN numStripes) . forM allocationWithIndex $ \(i, n) -> do
```

Reply:
oof, owned by bad variable names D: Renamed everything to be clearer